lib/Onyx.js

```diff
@@ -1106,11 +1096,9 @@ function clear(keysToPreserve = []) {
                 notifyCollectionSubscribersOnNextTick(key, value);
             });
 
-            // Call clear() and make sure that the default key/values and the key/values from the parameter
-            // are preserved in storage. This makes sure to always leave storage in a state that contains
-            // all the default values and any additional values that we want to remain after the database is cleared.
-            return Storage.clear()
-                .then(() => Storage.multiSet([...defaultKeyValuePairs, ...keyValuesToPreserve]));
+            // Remove only the items that we want cleared from storage
+            const retainedKeyValuePairs = [...defaultKeyValuePairs, ...keyValuesToPreserve];
+            return Storage.removeItems(keysToBeClearedFromStorage).then(() => Storage.multiSet(retainedKeyValuePairs));
```

Comment:
Do we still need to do a `multiSet`, given that we didn't remove any key/value pairs for the retained keys?

```suggestion
            return Storage.removeItems(keysToBeClearedFromStorage);
```

Reply:
Oh, I guess we do want to reset the default keys back to their default

Reply:
> Oh, I guess we do want to reset the default keys back to their default

We still reset the default keys back to default value

Reply:
Okay, so that means we should make changes like this:

```diff
diff --git a/lib/Onyx.js b/lib/Onyx.js
index ed93f15..9322e69 100644
--- a/lib/Onyx.js
+++ b/lib/Onyx.js
@@ -1050,8 +1050,6 @@ function clear(keysToPreserve = []) {
             const keysToBeClearedFromStorage = [];
             const keyValuesToResetAsCollection = {};
             const keyValuesToResetIndividually = {};
-            const keyValuesToPreserve = _.map(keysToPreserve, key => [key, cache.getValue(key)]);
-            const defaultKeyValuePairs = _.pairs(_.omit(defaultKeyStates, ...keysToPreserve));
 
             // The only keys that should not be cleared are:
             // 1. Anything specifically passed in keysToPreserve (because some keys like language preferences, offline
@@ -1096,9 +1094,10 @@ function clear(keysToPreserve = []) {
                 notifyCollectionSubscribersOnNextTick(key, value);
             });
 
-            // Remove only the items that we want cleared from storage
-            const retainedKeyValuePairs = [...defaultKeyValuePairs, ...keyValuesToPreserve];
-            return Storage.removeItems(keysToBeClearedFromStorage).then(() => Storage.multiSet(retainedKeyValuePairs));
+            // Remove only the items that we want cleared from storage, and reset others to default
+            const defaultKeyValuePairs = _.pairs(_.omit(defaultKeyStates, keysToPreserve));
+            return Storage.removeItems(keysToBeClearedFromStorage)
+                .then(() => Storage.multiSet(defaultKeyValuePairs));
         });
 }
```